go/database/mpt/nodes_test.go

```diff
@@ -6554,6 +6554,162 @@ func markModifiedAsDirty(t *testing.T, ctxt *nodeContext, before, after NodeRefe
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			ctxt := newNiceNodeContext(t, ctrl)
+
+			root, shared := ctxt.Build(test.trie)
+			accountPresent := false
+			handle := shared.GetViewHandle()
+			if _, err := handle.Get().Visit(ctxt, &root, 0, MakeVisitor(func(n Node, i NodeInfo) VisitResponse {
+				if node, ok := n.(*AccountNode); ok && node.address == address {
+					accountPresent = true
+				}
+				return VisitResponseContinue
+			})); err != nil {
+				t.Fatalf("unexpected error during final visit: %v", err)
+			}
+			handle.Release()
+
+			visitor := NewMockNodeVisitor(ctrl)
+			var last *gomock.Call
+			for i, label := range test.path {
+				// the visitor does not return incorrect accounts,
+				// i.e., include the last node in the path only when
+				// the account does exist.
+				if i < len(test.path)-1 || accountPresent {
+					ref, shared := ctxt.Get(label)
+					handle := shared.GetViewHandle()
+					cur := visitor.EXPECT().Visit(handle.Get(), NodeInfo{Id: ref.Id()})
+					handle.Release()
+					if last != nil {
+						cur.After(last)
+					}
+					last = cur
+				}
+			}
+
+			found, err := VisitPathToAccount(ctxt, &root, address, visitor)
+			if err != nil {
+				t.Fatalf("unexpected error during path iteration: %v", err)
+			}
+
+			if found != accountPresent {
+				t.Errorf("unexpected found result, wanted %t, got %t", accountPresent, found)
+			}
+		})
+	}
+}
+
+func TestVisitPathToAccount_SourceError(t *testing.T) {
+	injectedErr := errors.New("injected error")
+
+	ctrl := gomock.NewController(t)
+
+	source := NewMockNodeSource(ctrl)
+	source.EXPECT().getConfig().Return(S4LiveConfig).AnyTimes()
+	source.EXPECT().getViewAccess(gomock.Any()).Return(shared.MakeShared[Node](EmptyNode{}).GetViewHandle(), injectedErr)
+
+	nodeVisitor := NewMockNodeVisitor(ctrl)
+	nodeVisitor.EXPECT().Visit(gomock.Any(), gomock.Any()).AnyTimes()
+
+	var address common.Address
+	rootId := NewNodeReference(EmptyId())
+	if success, _ := VisitPathToAccount(source, &rootId, address, nodeVisitor); success {
+		t.Fatalf("expected iteration to fail")
+	}
+}
+
+func TestVisitPathToAccount_VisitorAborted(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	nodeVisitor := NewMockNodeVisitor(ctrl)
+	nodeVisitor.EXPECT().Visit(gomock.Any(), gomock.Any()).DoAndReturn(func(Node, NodeInfo) VisitResponse {
+		return VisitResponseAbort
+	}) // will be executed only once, then aborted
+
+	var address common.Address
+	extNode := ExtensionNode{path: CreatePathFromNibbles(addressToNibbles(address))}
+
+	source := NewMockNodeSource(ctrl)
+	source.EXPECT().getConfig().Return(S4LiveConfig).AnyTimes()
+
+	source.EXPECT().getViewAccess(gomock.Any()).Return(shared.MakeShared[Node](&extNode).GetViewHandle(), nil)
+
+	rootId := NewNodeReference(EmptyId())
+	if success, _ := VisitPathToAccount(source, &rootId, address, nodeVisitor); success {
+		t.Fatalf("expected iteration to fail")
+	}
```

Comment:
Could be reduced to this:
```suggestion
	ctrl := gomock.NewController(t)
	ctxt := newNiceNodeContext(t, ctrl)

	var address common.Address
	root, _ := ctxt.Build(&Extension{path: []Nibble{0}})

	nodeVisitor := NewMockNodeVisitor(ctrl)
	nodeVisitor.EXPECT().Visit(gomock.Any(), gomock.Any()).Return(VisitResponseAbort)

	if success, _ := VisitPathToAccount(ctxt, &root, address, nodeVisitor); success {
		t.Fatalf("expected iteration to fail")
	}
```